internal/app/plugins/kafka/kafka.go

```diff
@@ -0,0 +1,192 @@
+package kafka
+
+import (
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"strconv"
+	"time"
+
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+
+	"github.com/resonatehq/resonate/internal/aio"
+	"github.com/resonatehq/resonate/internal/kernel/t_aio"
+	"github.com/resonatehq/resonate/internal/metrics"
+)
+
+type Config struct {
+	Size        int           `flag:"size" desc:"submission buffered channel size" default:"100"`
+	Workers     int           `flag:"workers" desc:"number of workers" default:"1"`
+	Timeout     time.Duration `flag:"timeout" desc:"kafka request timeout" default:"30s"`
+	TimeToRetry time.Duration `flag:"ttr" desc:"time to wait before resending" default:"15s"`
+	TimeToClaim time.Duration `flag:"ttc" desc:"time to wait for claim before resending" default:"0"`
+	Brokers     []string      `flag:"brokers" desc:"kafka broker addresses" default:"localhost:9092"`
+	Compression string        `flag:"compression" desc:"compression type (none, gzip, snappy, lz4, zstd)" default:"none"`
+}
+
+type Producer interface {
+	Produce(msg *kafka.Message, deliveryChan chan kafka.Event) error
+	Flush(timeoutMs int) int
+	Close()
+}
+
+type Worker struct {
+	i        int
+	sq       <-chan *aio.Message
+	timeout  time.Duration
+	aio      aio.AIO
+	metrics  *metrics.Metrics
+	config   *Config
+	producer Producer
+}
+
+type Kafka struct {
+	sq      chan *aio.Message
+	workers []*Worker
+}
+
+type Addr struct {
+	Topic   string            `json:"topic"`
+	Key     *string           `json:"key,omitempty"`
+	Headers map[string]string `json:"headers,omitempty"`
+}
+
+func (a *Addr) validate() error {
+	if a.Topic == "" {
+		return fmt.Errorf("topic required")
+	}
+	return nil
+}
+
+func New(a aio.AIO, metrics *metrics.Metrics, config *Config) (*Kafka, error) {
+	producer, err := kafka.NewProducer(newConfig(config))
+	if err != nil {
+		return nil, fmt.Errorf("kafka producer: %w", err)
+	}
+	return NewWithProducer(a, metrics, config, producer)
+}
+
+func newConfig(config *Config) *kafka.ConfigMap {
+	c := &kafka.ConfigMap{
+		"bootstrap.servers":   fmt.Sprintf("%v", config.Brokers),
+		"delivery.timeout.ms": int(config.Timeout.Milliseconds()),
```

Comment:
I wonder if it's possible to convert the int64 here to a string instead of an int?